File: 0028. Find the Index of the First Occurrence in a String/solution_1.java
class Solution
{
    public int strStr(String haystack, String needle)
    {
        if (haystack.length() < needle.length())
            return -1;

        int index = -1;

        for (int i = 0; i <= haystack.length() - needle.length(); ++i)
        {
            boolean ok = true;

            for (int j = 0; j < needle.length(); ++j)
            {
                if (haystack.charAt(i + j) != needle.charAt(j))
                {
                    ok = false;
                    break;
                }
            }

            if (ok)
            {
                index = i;
                break;
            }

        }
        
        return index;
    }
}

File: 0028. Find the Index of the First Occurrence in a String/solution_1.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */

var strStr = function(haystack, needle) 
{
    let index = -1;
    
    for (let i = 0; i < haystack.length; ++i)
    {
        let ok = true;

        for (let j = 0; j < needle.length; ++j)
        {
            if (haystack[i + j] != needle[j])
            {
                ok = false;
                break;
            }
        }

        if (ok)
        {
            index = i;
            break;
        }
    }

    return index;
};